tests/unit/sql_test.py

```diff
@@ -282,11 +281,11 @@ def test_statement_cursor_buffer_size(self):
             statement.cursor_buffer_size = valid
             self.assertEqual(valid, statement.cursor_buffer_size)
 
-        invalid_inputs = [0, -10, -99999]
+        invalid_inputs = [0, -10, -99999, "hey", None, 1.0]
 
         for invalid in invalid_inputs:
             statement = SqlStatement("something")
-            with self.assertRaises(ValueError):
+            with self.assertRaises((ValueError, AssertionError)):
                 statement.cursor_buffer_size = invalid
 
     def test_statement_expected_result_type(self):
```

Comment:
I think 'ROWS' is also valid. Maybe we can add that too. BTW I think currently passing 'ROWS' to the expected result setter does not throw an error in setter but causes a problem when executing SQL statement. This because, as I see, try to get enum method accepts string versions of enums but it converts them to numbers.